src/matrices.jl

```diff
@@ -0,0 +1,53 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+abstract type CoDaMatrix{T} end
```

Comment:
I think we should remove this abstract parent type and make all elementary matrices a subtype of AbstractMatrix as we had before. That way Julia will have better defaults for the type, and instances will be accepted in interfaces written in 3rd-party packages.

Reply:
To make sizeless objects, we can't define the structure as an instance of `AbstractMatrix`. Just like `UniformScaling` from `LinearAlgebra.jl`, we have to make it a distinct type. Otherwise, `getindex` and `size` must be defined.

I forgot to leave a note to read the commit message: the main points were addressed there.